pkg/kotsadm/minio.go

```diff
@@ -21,10 +23,25 @@ import (
 	"k8s.io/client-go/kubernetes/scheme"
 )
 
+var (
+	MinioImageTagDateRegexp = regexp.MustCompile(`RELEASE\.(\d{4}-\d{2}-\d{2}T\d{2}-\d{2}-\d{2}Z)`)
+	// MigrateToMinioXlBeforeTime is the time that the minio version was released that removed the legacy backend
+	// that we need to migrate from: https://github.com/minio/minio/releases/tag/RELEASE.2022-10-29T06-21-33Z
+	MigrateToMinioXlBeforeTime = time.Date(2022, 10, 29, 6, 21, 33, 0, time.UTC)
+)
+
 func getMinioYAML(deployOptions types.DeployOptions) (map[string][]byte, error) {
 	docs := map[string][]byte{}
 	s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
 
+	if deployOptions.MigrateToMinioXl {
+		var configMap bytes.Buffer
+		if err := s.Encode(kotsadmobjects.MinioXlMigrationScriptsConfigMap(deployOptions.Namespace), &configMap); err != nil {
+			return nil, errors.Wrap(err, "failed to marshal minio migration configmap")
+		}
+		docs["minio-xl-migration-configmap.yaml"] = configMap.Bytes()
```

Comment:
do you not need the status configmap as well?

Reply:
that configmap is created, checked, and updated during `admin-cosole upgrade` only to prevent re-runs of the command from editing an in-progress or failed migration and removing initcontainers.  generated manifests won't have that same cli logic running.  though i suppose the user could re-generate and re-apply manifests during a migration to mess things up?

Reply:
ah good point. i don't think we need to worry about that edge case. thanks for the explanation!